java/arcs/core/storage/ReferenceModeStore.kt

```diff
@@ -184,6 +185,13 @@ class ReferenceModeStore private constructor(
     receiveQueue.enqueue { }
   }
 
+  override fun close() {
+    containerStore.close()
+    runBlocking {
```

Comment:
we are trying to avoid runBlocking. I am not sure what's the best way to do so here, but I think making close a suspend function is ok.

Also, I suspect it is better to run these in queue like Jibbl is doing here: https://critique-ng.corp.google.com/cl/339928017/depot/google3/third_party/java_src/arcs/java/arcs/core/storage/ReferenceModeStore.kt

Reply:
Yes, I think enqueueing the close operations on the receiveQueue is definitely the right thing to do here. Although we may still encounter a problem if more operations are queued while the close is pending...

In this specific case, I don't think the runBlocking will cause us problems, but that could easily change in the future. Other options that come to mind:

* change the queue implementation so that the enqueue method isn't suspending. I don't think it's possible with the current implementation, but if we replace it later with a channel-based implementation then it will be an easy change.

* the `create` method takes a coroutineScope parameter. We could provide the coroutineScope to the ReferenceModeStore as well, and then that scope could be used to do the enqueue here, for now. This is assuming it's OK if the close method exits before the close has completed.

Reply:
Making the close method suspending is a nice option, but the current DirectStoreMuxer impl requires a non-suspending close method. This could be fixed, though.

Reply:
For now, passing in the scope and using that seems like the best approach.